sqlalchemy_redshift/dialect.py

```diff
@@ -689,8 +689,11 @@ def _get_all_column_info(self, connection, **kw):
             # replace the original value for search_path.
             schema_names = ['"%s"' % r.name for r in cc.execute("""
             SELECT nspname AS "name"
-            FROM pg_catalog.pg_namespace
-            WHERE nspname !~ '^pg_' AND nspname <> 'information_schema'
+            FROM pg_catalog.pg_namespace n
+            WHERE nspname !~ '^pg_'
+                AND nspname <> 'information_schema'
+                AND n.oid NOT IN
+                  (SELECT esoid FROM pg_catalog.pg_external_schema)
```

Comment:
Tests against Redshift were failing for this. It looks like `schema_names` was returning an empty string every time. Pushed this change, moving to a check based on membership in an array populated from a subquery.